src/bootstrap/src/core/download.rs

```diff
@@ -250,19 +257,70 @@ impl Config {
                             "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')",
                             url, tempfile.to_str().expect("invalid UTF-8 not supported with powershell downloads"),
                         ),
-                    ])).is_err() {
+                    ])).is_ok() {
                         return;
                     }
                     eprintln!("\nspurious failure, trying again");
                 }
             }
             if !help_on_error.is_empty() {
                 eprintln!("{help_on_error}");
+                Self::check_outdated(commit);
             }
             crate::exit!(1);
         }
     }
 
+    fn check_outdated(commit: &String) {
+        let last_commit: String = String::from_utf8(
+            Command::new("git")
+                .arg("show")
+                .arg("-s")
+                .arg("--date=short")
+                .arg("--format=%ct")
+                .arg("HEAD")
+                .output()
+                .expect("Failed to get git log")
+                .stdout,
+        )
+        .unwrap_or("".to_string());
+        match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(n) => {
+                let replaced = last_commit.replace("\n", "");
```

Comment:
```suggestion
                let replaced = last_commit.trim();
```

I suspect this is the actual intent.